### PR TITLE
fix(security): resolve remaining code scanning alerts

### PR DESCRIPTION
**Pull request by JanPokorny**
## Summary

Resolves 36 of the 38 remaining open code scanning alerts. The 2 left are:
- **#15** (network data to file) — false positive; intentional tarball download with size validation
- **#150, #151** (workflow permissions) — already filed as PR #274

**High severity (8 → 0):**
- **#17** — TOCTOU in `local-skill-repository.ts`: use file handle so `stat`+`read` operate on the same inode
- **#18** — TOCTOU in `server.ts`: remove `existsSync` guard, just try/catch `readFileSync`
- **#14** — Missing regex anchor in `skills-panel.tsx`: add `\b` word boundaries to prevent `evilgithub.com` matching
- **#19-23** — Empty password in `values.yaml`: change defaults from `""` to `null` (templates use `default (randAlphaNum 32)`, so behavior is identical)

**Medium severity (28 → 0):**
- **#78-105** — Code injection in `maximize-build-space/action.yml`: pass all `${{ inputs.* }}` through `env:` block instead of inline shell interpolation

## Test plan

- [x] `mise run check` passes (lint + typecheck + helm lint + kubeconform)
- [x] `mise run test` passes (575 tests across all packages)

### CODE REVIEW

File: packages/ui/src/modules/sessions/components/skills-panel.tsx [hunk 1]
```diff
@@ -88,7 +88,10 @@ function skillSourceUrl(
   compareFrom?: string,
 ): string {
   const base = source.replace(/\.git$/, "").replace(/\/$/, "");
-  const isGitLike = /(github|gitlab)\.com|bitbucket\.org/.test(base);
+  const isGitLike =
+    /^(?:https?:\/\/)?(?:www\.)?(?:github\.com|gitlab\.com|bitbucket\.org)(?::\d+)?(?:\/|$)/.test(
+      base,
+    );
```
Comment (xjacka) [inline packages/ui/src/modules/sessions/components/skills-panel.tsx:94]: 🟢 **Suggestion:** Consider adding a unit test for `skillSourceUrl` covering the security-relevant edge cases of this new regex — e.g. `evilgithub.com` → not gitLike, `github.com.evil.org` → not gitLike, `https://github.com/owner/repo` → gitLike. This would prevent future regressions from silently re-introducing the substring-match vulnerability.

**Review by xjacka — APPROVED** [review]
🛡️ **DAM** — Code Review @ `1d63833c`

## PR #275: fix(security): resolve remaining code scanning alerts
**Author:** JanPokorny | **Branch:** fix/e2e-permissions#locki-zwru5lds → main | **Changes:** +83 −62 (6 files)

### Summary
Resolves 36 of 38 open CodeQL alerts: TOCTOU races in `local-skill-repository.ts` (file handle pattern) and `server.ts` (try/catch replaces existsSync guard), empty-string password defaults replaced with `null` in Helm values, git-host detection regex anchored in `skills-panel.tsx`, and GitHub Actions workflow permissions locked down. The two deferred alerts are a confirmed false positive (#15) and workflow permissions covered by the now-merged PR #274.

### Findings
- ✅ **Looks good:** TOCTOU fix in `local-skill-repository.ts` — `open()+fh.stat()+fh.readFile()` ensures stat and read operate on the same inode; `finally` guarantees `fh.close()` even on early returns. (`packages/agent-runtime/src/modules/skills/infrastructure/local-skill-repository.ts:142`)
- ✅ **Looks good:** TOCTOU fix in `server.ts` — `try/catch` replaces `existsSync`+`try/catch`, eliminating the stat/read race with no observable behavior change. (`packages/agent-runtime/src/server.ts:304`)
- ✅ **Looks good:** All 13 inputs in `maximize-build-space/action.yml` correctly moved to the `env:` block; no inline `${{ inputs.* }}` interpolation remains.
- ✅ **Looks good:** `skills-panel.tsx` regex uses a full anchored URL pattern, exceeding CodeQL's `\b`-boundary suggestion and preventing `github.com.evil.org` bypass.
- 🟢 **Suggestion:** Add a unit test for `skillSourceUrl` covering the new regex's security edges (`evilgithub.com` → not gitLike, `github.com.evil.org` → not gitLike, valid `https://github.com/owner/repo` → gitLike) to lock in the fix against future regressions. (`packages/ui/src/modules/sessions/components/skills-panel.tsx:91`)

### Documentation Check (doc-drift)
✅ No documentation drift detected.

### Verdict
APPROVE — All six security fixes are technically sound, the TOCTOU patterns are correct, and the regex anchoring goes beyond the bot's own suggestion in robustness.

---
_Review by [DAM](https://dam.ai) · automated code guardian_
